cobalt/migrate_storage_record/migrate.cc

```diff
@@ -0,0 +1,398 @@
+         local_storages.local_storage_entries()) {
+      entries[local_storage_entry.key()] = local_storage_entry.value();
+    }
+  }
+  if (entries.size() == 0) {
+    std::move(callback).Run(std::move(storage_data));
+    return;
+  }
+  std::atomic_int* pending = new std::atomic_int();
+  pending->store(entries.size());
+  base::OnceClosure* all_done_callback = new base::OnceClosure(
+      base::BindOnce(std::move(callback), std::move(storage_data)));
+  for (const auto& entry : entries) {
+    content::RenderFrameHost::JavaScriptResultCallback
+        javascript_result_callback = base::BindOnce(
+            [](std::atomic_int* pending, base::OnceClosure* all_done_callback,
+               base::Value result) {
+              --(*pending);
+
+              if (pending->load() == 0) {
+                delete pending;
+                std::move(*all_done_callback).Run();
+                delete all_done_callback;
+              }
+            },
+            pending, all_done_callback);
+    std::string object_name = "localStorage";
+    std::string method_name = "setItem";
+    std::string key = entry.first;
+    std::string value = entry.second;
+    base::Value::List arguments = base::Value::List().Append(key).Append(value);
+    render_frame_host->ExecuteJavaScriptMethod(
+        base::ASCIIToUTF16(object_name), base::ASCIIToUTF16(method_name),
+        std::move(arguments), std::move(javascript_result_callback));
+  }
+}
+
+void ClearAllCookies(content::RenderFrameHost* render_frame_host,
+                     base::OnceClosure callback) {
+  LOG(INFO) << "Clearing cookies.";
+  content::StoragePartition* storage_partition =
+      render_frame_host->GetStoragePartition();
+  DCHECK(storage_partition);
+  if (!storage_partition) {
+    return;
+  }
+  network::mojom::CookieManager* cookie_manager =
+      storage_partition->GetCookieManagerForBrowserProcess();
+  DCHECK(cookie_manager);
+  if (!cookie_manager) {
+    return;
+  }
+  cookie_manager->DeleteCookies(
+      network::mojom::CookieDeletionFilter::New(),
+      base::BindOnce([](base::OnceClosure callback,
+                        uint32_t num_deleted) { std::move(callback).Run(); },
+                     std::move(callback)));
+}
+
+void ClearAllLocalStorage(content::RenderFrameHost* render_frame_host,
+                          base::OnceClosure callback) {
+  LOG(INFO) << "Clearing localStorage.";
+  std::string object_name = "localStorage";
+  std::string method_name = "clear";
+  base::Value::List arguments;
+  render_frame_host->ExecuteJavaScriptMethod(
+      base::ASCIIToUTF16(object_name), base::ASCIIToUTF16(method_name),
+      std::move(arguments),
+      base::BindOnce([](base::OnceClosure callback,
+                        base::Value result) { std::move(callback).Run(); },
+                     std::move(callback)));
+}
+
+}  // namespace
+
+void Migrate(content::RenderFrameHost* render_frame_host) {
+  const auto& start = base::TimeTicks::Now();
```

Comment:
We shouldn't do this just for the INFO (WARNING?) purposes below in l.364.

If you were absolutely into it, you should use ElapsedTimer instead (https://source.chromium.org/chromium/chromium/src/+/main:base/timer/elapsed_timer.h;bpv=0;bpt=1) which is more semantically accurate.